File: 01_Language/01_Functions/js/array_change_key_case.js
function array_change_key_case(arr, caseType = 'CASE_LOWER') {
    if (Array.isArray(arr)) {
        return arr;
    }

    let res = {}
    let func = (!caseType || caseType === 'CASE_LOWER') ? 'toLowerCase' : 'toUpperCase'
    for (let k in arr) {
        res[k[func]()] = arr[k]
    }
    return res
}

let obj = {
    "FirSt": 1,
    "SecOnd": 4,
}
console.log(array_change_key_case(obj, "CASE_UPPER"))
console.log(array_change_key_case([1, 3, 5, 7, 9], "CASE_UPPER"))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/ArrayChangeKeyCase.java
package com.onevgo.functions;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.stream.Collectors;

public class ArrayChangeKeyCase {
    public static enum CaseType {
        CASE_LOWER, CASE_UPPER
    }

    public static <V> Map<String, V> arrayChangeKeyCase(Map<String, V> input) {
        return arrayChangeKeyCase(input, CaseType.CASE_LOWER);
    }

    public static <V> Map<String, V> arrayChangeKeyCase(Map<String, V> input, CaseType caseType) {
        return input.entrySet().stream().collect(Collectors.toMap(e -> {
            if (caseType == null || caseType == CaseType.CASE_LOWER) {
                return e.getKey().toLowerCase();
            } else {
                return e.getKey().toUpperCase();
            }
        }, Map.Entry::getValue, (o, o2) -> o2, LinkedHashMap::new));
    }

    public static void main(String[] args) {
        Map<String, Object> map = new LinkedHashMap<>();
        map.put("FirSt", 1);
        map.put("FirST", 2);
        map.put("SecOnd", 4);
        System.out.println(arrayChangeKeyCase(map, CaseType.CASE_UPPER));
    }
}
